dotnet/src/Microsoft.Agents.AI.Declarative/ChatClient/ChatClientAgentFactory.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.ObjectModel;
+using Microsoft.Extensions.AI;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Agents.AI;
+
+/// <summary>
+/// Provides an <see cref="AgentFactory"/> which creates instances of <see cref="ChatClientAgent"/>.
+/// </summary>
+public sealed class ChatClientAgentFactory : AgentFactory
```

Comment:
This factory is quite generalistic to be in this package. Shouldn't this be in the core instead or otherwise be `internal` and created solely via `extension` ways?

```suggestion
internal sealed class ChatClientAgentFactory : AgentFactory
```

Reply:
It cannot be `internal` because then developers cannot use it. If I move this to the core package then I also need to add the following dependencies

```
    <PackageReference Include="Microsoft.Bot.ObjectModel" />
    <PackageReference Include="Microsoft.Bot.ObjectModel.Json" />
    <PackageReference Include="Microsoft.Bot.ObjectModel.PowerFx" />
    <PackageReference Include="Microsoft.PowerFx.Interpreter" />
```